Task-56509: Configure the notes Application as a main page application

Amine0008 opened the pull request:
This change will allow to capture the main application name. In fact, any application will be able to define the variable eXo.env.portal.mainApplicationName (with the application/module name) inside its required JS in order to add an additional data property inside collected operation pageFullUIDisplay

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

Amine0008 committed:
added the mainApplicationName when creating the notes application

boubaker reviewed (commented):
Please update title & description as suggested in other PRs and add application name label inside Analytics_en.properties as suggested here https://github.com/Meeds-io/social/pull/1435#discussion_r858927118

hakermi committed:
TASK-56675: Deleted notes still accessible by note page links (#503)

Prior to this change,when deleteing a notes its deleted flag is set to true, while when getting the note by its id not control is added to check if it's deleted or not, and so it's displayed to the user,
This PR should add a check on the dleted flag and return NOT_FOUND http state code if it's deleted

hakermi committed:
TASK-56665: Notes drafts cannot be displayed in the notes tree (#505)

Prior to this change, when selecting drafts filter in notes tree drawer, there was an issue due to an OutOfBoundArrayException because a bad code placement when filter notes children depends on
the withDrafts query param, which selecting the child index in the list of children and then filter the list and override its size while keeping and using the old index to insert the draft child in the list
This PR should fix this problem by correctly place the code of fitring the list before selecting the child index to avoid having wrong index and an exception

azayati committed:
Task-55805: [Accessibility - Contrast - 3.2] Change the default primary color (#507)

Amine0008 committed:
changed the mainApplicationName property from Notes to Note so that it can be translated using the analytics.Note property
